app/gui/src/dashboard/hooks/searchParamsStateHooks.ts

```diff
@@ -40,7 +40,8 @@ export function useSearchParamsState<T = unknown>(
   defaultValue: T | (() => T),
   predicate: (unknown: unknown) => unknown is T = (unknown): unknown is T => true,
 ): SearchParamsStateReturnType<T> {
-  const { router, searchParams } = useRouterInReact()
+  const { router, searchParams: searchParamsRaw } = useRouterInReact()
+  const searchParams = React.useDeferredValue(searchParamsRaw)
```

Comment:
Why did you add 'useDefferedValue' here? It exists for a different purpose

Reply:
Different purpose than what?

Reply:
You clearly _don't_ need it here because the only purpose for this hook is _deferring_ rendering for the parts of the UI.  Why do you want to _defer_ here?

Reply:
When changing category or current id (the state being kept in search params) we want to defer updating asset table. I thought that it should be the default for every query update. But perhaps you're right, I should use it near actual suspense queries.

Reply:
In that case we should wrap category change in 'startTransition'

Deferred state has completely different purpose compared to 'startTransition'

Reply:
> Deferred state has completely different purpose compared to 'startTransition'

Didn't know that. I was just reading in [`startTransition` docs](https://react.dev/reference/react/startTransition#caveats), that:

> You can wrap an update into a Transition only if you have access to the set function of that state. If you want to start a Transition in response to some prop or a custom Hook return value, try [useDeferredValue](https://react.dev/reference/react/useDeferredValue) instead.

ATM the category is based on the context change coming from vue.